lib/model/message.dart

```diff
@@ -137,21 +539,30 @@ class MessageStoreImpl extends PerAccountStoreBase with MessageStore {
     //   [InheritedNotifier] to rebuild in the next frame) before the owner's
     //   `dispose` or `onNewStore` is called.  Discussion:
     //     https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/MessageListView.20lifecycle/near/2086893
+
+    assert(!_disposed);
+    _disposeOutboxMessages();
+    _disposed = true;
   }
 
   @override
   Future<void> sendMessage({required MessageDestination destination, required String content}) {
-    // TODO implement outbox; see design at
-    //   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M3881.20Sending.20outbox.20messages.20is.20fraught.20with.20issues/near/1405739
-    return _apiSendMessage(connection,
-      destination: destination,
-      content: content,
-      readBySender: true,
-    );
+    assert(!_disposed);
+    if (!debugOutboxEnable) {
+      return _apiSendMessage(connection,
+        destination: destination,
+        content: content,
+        readBySender: true);
+    }
+    return _outboxSendMessage(
+      destination: destination, content: content,
+      // TODO move [TopicName.processLikeServer] to a substore, eliminating this
+      realmEmptyTopicDisplayName: _realmEmptyTopicDisplayName);
```

Comment:
nit: let's add a link to https://github.com/zulip/zulip-flutter/pull/1472#discussion_r2099069276 to expand on the details of this thought:
```suggestion
      // TODO move [TopicName.processLikeServer] to a substore, eliminating this;
      //   see https://github.com/zulip/zulip-flutter/pull/1472#discussion_r2099069276
      realmEmptyTopicDisplayName: _realmEmptyTopicDisplayName);
```